src/main/scala/com/fasterxml/jackson/module/scala/modifiers/ScalaTypeModifier.scala

```diff
@@ -0,0 +1,33 @@
+package com.fasterxml.jackson.module.scala.modifiers
+
+import java.lang.reflect.{ParameterizedType, Type}
+
+import com.fasterxml.jackson.databind.JavaType
+import com.fasterxml.jackson.databind.`type`._
+import com.fasterxml.jackson.module.scala.JacksonModule
+
+class ScalaTypeModifier extends TypeModifier {
+  override def modifyType(javaType: JavaType,
+                          jdkType: Type,
+                          context: TypeBindings,
+                          typeFactory: TypeFactory): JavaType = {
+
+
+    if (javaType.isTypeOrSubTypeOf(classOf[collection.Map[_, _]])) {
+      MapLikeType.upgradeFrom(javaType, javaType.containedTypeOrUnknown(0), javaType.containedTypeOrUnknown(1))
+    } else if (javaType.isTypeOrSubTypeOf(classOf[collection.TraversableOnce[_]])) {
+      CollectionLikeType.upgradeFrom(javaType, javaType.containedTypeOrUnknown(0))
+    } else if (javaType.isTypeOrSubTypeOf(classOf[Option[_]])) {
+      ReferenceType.upgradeFrom(javaType, javaType.containedTypeOrUnknown(0))
```

Comment:
This should actually be simplified to only do direct comparison to `Option`, and not subtypes, since as of 2.7 type resolution works sort of layer by layer. In most cases it won't matter a lot, but for custom subtypes (if one wants to bind types for convenience, like `FoobarOption` or something) it works better.